### ISSUE TITLE
Regression: MySQL Query Builder does not filter by selected time/date range

### ISSUE THREAD

**Issue by Jay2k1** [grafana/grafana#68750]
### 
# What went wrong?

After quite a while, I decided to add a new (time series) graph based on MySQL data. I was surprised to find myself unable to "click myself a new graph" in 9.5 as the query it comes up with looks like this:
```sql
SELECT
  inserted AS "time",
  temperature
FROM
  weather.meas_offset
WHERE
  NAME = 'Balkon'
LIMIT
  50
```
...leading my database to return four hours worth of data from 2019. This leads to a `Data outside time range` message in the graph. If I was a new user, I'd be clueless how to make grafana pull the data required for the currently selected time range (for me, defaulting to last 24 hours).

Only due to the fact that I already have some older graphs and am able to look at their (generated) SQL, I can basically copy-paste from there - an example:

```sql
SELECT
  $__timeGroupAlias(inserted,$__interval,NULL),
  avg(temperature) AS "Temperatur"
FROM meas_offset
WHERE
  $__timeFilter(inserted) AND
  name = 'Balkon'
GROUP BY 1
ORDER BY $__timeGroup(inserted,$__interval,NULL)
```
I was looking for a button to add these or a help text telling me what to do, but couldn't find any. But honestly, I expect grafana to automatically add these, as it has done for the last years that I have been using it. I am not a novice, but had I just installed grafana for the first time, I would probably have given up.

I then tried finding hints as for how to fix the query in the [docs](https://grafana.com/docs/grafana/latest/datasources/mysql/#time-series-queries), where I found a mention of using `WHERE $__timeFilter(time_date_time)`, but I couldn't figure out how to add this filter using the query builder. I had to switch to the `Code` panel, remove the `LIMIT 50` and add `WHERE $__timeFilter(time_date_time)` myself to get some useful data. But again, adding a simple time series graph should not require reading docs.

### 
# How do we reproduce it?

have a MySQL datasource with time series data (going back a while, maybe a year) and try to make a graph from it. Notice the UX difference between doing it with whatever grafana version had the old query builder and the 9.5.2 one.

### 
# What Grafana version are you using?

Grafana v9.5.2 (cfcea75916)

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

A package manager (APT, YUM, BREW, etc.)

### User's OS?

Ubuntu 22.04

### User's Browser?

Google Chrome Version 113.0.5672.92 (Official Build) (x86_64)

### Is this a Regression?

Yes

### Are Datasources involved?

MySQL

**Comment by mellieA** [comment]
@Jay2k1 thanks for opening this. Could you include the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here? It will create a snapshot where you can sanitize the data so we can reproduce the issue quickly.

**Comment by mellieA** [comment]
@Jay2k1 I was able to get a time series dataset to test with.

In my test I see that the date filter keeps toggling off when I apply or run the query and it's not intuitive how to build the query.

Dev squad, I tested this out on https://play.grafana.org/d/f9Oiow97k/mysql-8-sakila-video-store-dummy-data?orgId=1&editPanel=2 

Grafana 9.5.3

To test:
- Attempt to reproduce the query stored in this panel using only the builder mode and the drop downs
- Turn on the filter toggle for the date filter
- Click **Run Query** or **Apply** and notice that the filter toggle turns back off

**Comment by Jay2k1** [comment]
Thanks for taking care of this!

> In my test I see that the date filter keeps toggling off when I apply or run the query and it's not intuitive how to build the query.

**My main concern** is that there does not seem to be a date filter in the query builder that adds the required macros to filter by **currently selected date range**. I could probably use the filter dropdowns to apply a fixed filter with absolute dates, but that's not what I need - I need the graphs to show the data of whatever time range I select in the top right time range picker (last X hours/days/months) and I opened this bug because I cannot figure out how to build such a query/filter from scratch.

FWIW, this seems to be known: #53053 from August '22 mentions `A filter for the current selected timerange is automatically added to new queries.` in the list of missing features after the Angular ➜ React rewrite.

I understand you found out that trying to use the date filter at all does not work and it toggles off when you try to use it; I could not reproduce that (see screenshot below), but if this is bugged, it's another issue.

![image](https://github.com/grafana/grafana/assets/2070503/ee6c7755-dba8-4f73-b5f5-dc2d4c10c5df)

On your linked playground panel, I added an arbitrarily picked time filter and clicked "Run query" and it stayed on.

**Comment by mdvictor** [comment]
Hey @Jay2k1. Thank you for opening this issue.

I agree that we need an easier way to add a filter for the selected time range, and as you pointed out, we are aware of the issue and working on improving the new query builder. I cannot give you a concrete date when this will be fixed, but I can tell you that need for improvements are under our radar.

**Comment by zoltanbedi** [comment]
This should also be addressed in https://github.com/grafana/grafana/issues/53053

**Comment by YerBabiE** [comment]
I can also confirm the same issue running the latest container build (Docker) on a Synology NAS. I have a standard mysql plug in and its not adding any ranges to sql querys.... It also seems to be adding a 50 point data limit as shown above. It was all working fine on previous editions....

**Comment by zoltanbedi** [comment]
What is left in this issue is adding support for `$__timeGroup` and `$__timeGroupAlias` macros.

**Comment by laundmo** [comment]
It's been a few months again and still no updates on this or #53053

Has anyone previously using Grafana for this simply jumped ship to a competitor at this point?

**Comment by bobemoe** [comment]
@laundmo the macros are available again in 10.3.0 now https://github.com/grafana/grafana/pull/74575 is merged, just not selected by default.

**Comment by gabor** [comment]
hi, as the original topic ( filter by time range ) has been added to grafana, i created a new issue that talks about the rest of the topic ( support for time-group and time-group-alias) at https://github.com/grafana/grafana/issues/83593 .
